### PR TITLE
Spring 05

### PR DESCRIPTION
**Pull request by Alex-Korzun**

### CODE REVIEW

**Review by Ketch92 — APPROVED** [review]
Great job!

File: src/main/java/com/dev/cinema/controller/CinemaHallController.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    private final CinemaHallMapper cinemaHallMapper;
+
+    @Autowired
+    public CinemaHallController(CinemaHallService cinemaHallService,
+                                CinemaHallMapper cinemaHallMapper) {
+        this.cinemaHallService = cinemaHallService;
+        this.cinemaHallMapper = cinemaHallMapper;
+    }
+
+    @PostMapping()
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controller/CinemaHallController.java]:
```suggestion
    @PostMapping
```
we can write as follows

File: src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 2]
```diff
@@ -63,4 +64,60 @@ public MovieSession add(MovieSession movieSession) {
...
+                    + movieSession, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public MovieSession delete(Long id) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java]: I am not sure if we need the entity back here
I think we can return void

File: src/main/java/com/dev/cinema/model/dto/request/MovieSessionRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
+package com.dev.cinema.model.dto.request;
+
+public class MovieSessionRequestDto {
+    private Long id;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/request/MovieSessionRequestDto.java]: do we need this field?

File: src/main/java/com/dev/cinema/service/mapper/CinemaHallMapper.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import com.dev.cinema.model.dto.response.CinemaHallResponseDto;
+import org.springframework.stereotype.Component;
+
+@Component
+public class CinemaHallMapper {
+    public CinemaHallResponseDto toDto(CinemaHall cinemaHall) {
+        CinemaHallResponseDto cinemaHallResponseDto = new CinemaHallResponseDto();
+        cinemaHallResponseDto.setId(cinemaHall.getId());
+        cinemaHallResponseDto.setCapacity(cinemaHall.getCapacity());
+        cinemaHallResponseDto.setDescription(cinemaHallResponseDto.getDescription());
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/mapper/CinemaHallMapper.java]: is everything ok on this line?

File: src/main/java/com/dev/cinema/service/mapper/MovieMapper.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import com.dev.cinema.model.dto.response.MovieResponseDto;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MovieMapper {
+    public MovieResponseDto toDto(Movie movie) {
+        MovieResponseDto movieResponseDto = new MovieResponseDto();
+        movieResponseDto.setId(movie.getId());
+        movieResponseDto.setTitle(movieResponseDto.getTitle());
+        movieResponseDto.setDescription(movieResponseDto.getDescription());
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/mapper/MovieMapper.java]: same here

File: src/main/java/com/dev/cinema/service/mapper/MovieMapper.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+        return movieResponseDto;
+    }
+
+    public Movie fromDto(MovieRequestDto movieRequestDto) {
+        Movie movie = new Movie();
+        movie.setTitle(movieRequestDto.getTitle());
+        movie.setDescription(movie.getDescription());
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/mapper/MovieMapper.java]: same

File: src/main/java/com/dev/cinema/service/mapper/MovieSessionMapper.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        movieSessionResponseDto.setMovieTitle(movieSession.getMovie().getTitle());
+        movieSessionResponseDto.setShowTime(movieSession.getShowTime().toString());
+        movieSessionResponseDto.setCinemaHallId(movieSession.getCinemaHall().getId());
+        movieSessionResponseDto.setCinemaHallCapacity(movieSession.getCinemaHall().getCapacity());
+        return movieSessionResponseDto;
+    }
+
+    public MovieSession fromDto(MovieSessionRequestDto movieSessionRequestDto) {
+        MovieSession movieSession = new MovieSession();
+        movieSession.setId(movieSessionRequestDto.getId());
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/mapper/MovieSessionMapper.java]: should we remove this line?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/controller/CinemaHallController.java (hunk 1 above)
Comment (Alex-Korzun) [reply src/main/java/com/dev/cinema/controller/CinemaHallController.java]: removed everywhere

File: src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java (hunk 2 above)
Comment (Alex-Korzun) [reply src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java]: agree. made it void.

File: src/main/java/com/dev/cinema/model/dto/request/MovieSessionRequestDto.java (hunk 3 above)
Comment (Alex-Korzun) [reply src/main/java/com/dev/cinema/model/dto/request/MovieSessionRequestDto.java]: removed `id` from `MovieSessionRequestDto`

File: src/main/java/com/dev/cinema/service/mapper/CinemaHallMapper.java (hunk 4 above)
Comment (Alex-Korzun) [reply src/main/java/com/dev/cinema/service/mapper/CinemaHallMapper.java]: was inattentive. changed getting description from Cinema Hall object.

File: src/main/java/com/dev/cinema/service/mapper/MovieMapper.java (hunk 5 above)
Comment (Alex-Korzun) [reply src/main/java/com/dev/cinema/service/mapper/MovieMapper.java]: changed

File: src/main/java/com/dev/cinema/service/mapper/MovieMapper.java (hunk 6 above)
Comment (Alex-Korzun) [reply src/main/java/com/dev/cinema/service/mapper/MovieMapper.java]: changed to movieRequestDto

File: src/main/java/com/dev/cinema/service/mapper/MovieSessionMapper.java (hunk 7 above)
Comment (Alex-Korzun) [reply src/main/java/com/dev/cinema/service/mapper/MovieSessionMapper.java]: yes, setting `id` is unnecessary for `RequestDto`

File: src/main/java/com/dev/cinema/service/mapper/MovieMapper.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+import com.dev.cinema.model.dto.request.MovieRequestDto;
+import com.dev.cinema.model.dto.response.MovieResponseDto;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MovieMapper {
+    public MovieResponseDto toDto(Movie movie) {
+        MovieResponseDto movieResponseDto = new MovieResponseDto();
+        movieResponseDto.setId(movie.getId());
+        movieResponseDto.setTitle(movieResponseDto.getTitle());
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/mapper/MovieMapper.java]:
```suggestion
        movieResponseDto.setTitle(movie.getTitle());
```

**Review by KseniiaMakarova — APPROVED** [review]

File: src/main/java/com/dev/cinema/service/mapper/MovieMapper.java (hunk 8 above)
Comment (Alex-Korzun) [reply src/main/java/com/dev/cinema/service/mapper/MovieMapper.java]: done

**Pull request merged**